### PR TITLE
[libc] Add `dlfcn.h` headers

### PR DESCRIPTION
**Pull request by izaakschroeder**
Does what it says on the box. Split out per request from @SchrodingerZhu

### CODE REVIEW

File: libc/spec/posix.td [hunk 1]
```diff
@@ -222,6 +222,40 @@ def POSIX : StandardSpec<"POSIX"> {
...
+    [
+      Macro<"RTLD_LAZY">,
+      Macro<"RTLD_NOW">,
+      Macro<"RTLD_GLOBAL">,
+      Macro<"RTLD_LOCAL">,
+    ],
+    [],  // Types
+    [], // Enumerations
+    [
+      FunctionSpec<
```
Comment (SchrodingerZhu) [inline libc/spec/posix.td:236]: Maybe leave function specs to the implementation part.

File: libc/include/llvm-libc-macros/CMakeLists.txt [hunk 2]
```diff
@@ -277,3 +277,9 @@ add_macro_header(
   HDR
     stdckdint-macros.h
 )
+
+add_macro_header(
+  dlfcn_macros
+  HDR
+    dlfcn-macros.h
+)
```
Comment (SchrodingerZhu) [inline libc/include/llvm-libc-macros/CMakeLists.txt:285]: add a trailing new line

File: libc/spec/posix.td (hunk 1 above)
Comment (izaakschroeder) [reply libc/spec/posix.td]: 👍

File: libc/include/llvm-libc-macros/CMakeLists.txt (hunk 2 above)
Comment (izaakschroeder) [reply libc/include/llvm-libc-macros/CMakeLists.txt]: 👍

File: libc/src/CMakeLists.txt [hunk 3]
```diff
@@ -11,6 +11,7 @@ add_subdirectory(stdio)
 add_subdirectory(stdlib)
 add_subdirectory(string)
 add_subdirectory(wchar)
+add_subdirectory(dlfcn)
```
Comment (lntue) [inline libc/src/CMakeLists.txt:14]: Look like this will have to go to the other PR.

Comment (izaakschroeder) [reply libc/src/CMakeLists.txt]: 🎉

**Review by SchrodingerZhu — APPROVED** [review]

**Pull request merged**